src/storage/src/source/postgres/snapshot.rs

```diff
@@ -197,23 +197,60 @@ pub(crate) fn render<G: Scope<Timestamp = MzOffset>>(
 
     let is_snapshot_leader = config.responsible_for("snapshot_leader");
 
-    // A filtered table info containing only the tables that this worker should read
-    let mut reader_table_info = table_info.clone();
-    reader_table_info.retain(|oid, _| config.responsible_for(oid));
+    // Determine which exports need to be snapshot, i.e. their offsets are 0.
+    let mut exports_to_snapshot: BTreeSet<_> = export_resume_uppers
+        .into_iter()
+        .filter_map(|(id, upper)| {
+            if upper.into_option() == Some(MzOffset::minimum()) {
+                // export_resume_uppers includes the remap collection, which source exports does
```

Comment:
That looks like a bug of the source ingestion framework. As a source implementor I shouldn't have to care that there are other shards that are being written. The parameter passed to `SourceRender` should have only the relevant frontiers for the source implementation

Reply:
Fixed here [storage: remove remap collection from resumption frontier calc](https://github.com/MaterializeInc/materialize/pull/18852/commits/663a851707cf0b523e5c06d57413e77224c4731f)